enhancements/network/baremetal-networking.md

```diff
@@ -0,0 +1,260 @@
+---
+title: in-cluster-network-infrastructure
+authors:
+  - "@jcpowermac"
+reviewers:
+  - "@abhinavdahiya"
+approvers:
+  - "@abhinavdahiya"
+creation-date: 2019-12-10
+last-updated:  2019-12-12
+status: implemented
+see-also:
+  - "https://github.com/openshift/installer/blob/master/docs/design/baremetal/networking-infrastructure.md"
+  - "https://github.com/openshift/installer/blob/master/docs/design/openstack/networking-infrastructure.md"
+  - "https://github.com/openshift/enhancements/pull/61"
+replaces:
+  - ""
+superseded-by:
+  - ""
+---
+
+# In Cluster Network Infrastructure
+or internal network services
+or internal networking infrastructure
+or non-cloud network{,ing} {services,infrastructure}
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Many customers still have on-premise infrastructure which includes bare metal, VMware vSphere,
+OpenStack and Red Hat RHV.  These customers would like to use the IPI installation approach while utilizing
+their existing environments. This enhancement would provide networking infrastructure including DNS
+and load balancing required for OpenShift to compute environments that do not provide such services.
+
+## Motivation
+
+### Goals
+
+Install an IPI OpenShift cluster on various on-premise non-cloud platforms that
+provides internal DNS and load balancing that is minimally required for OpenShift
+cluster to run properly.
+
+### Non-Goals
+
+## Proposal
+
+The basis and significant portions for this proposal was taken from existing
+[documentation](https://github.com/openshift/installer/blob/master/docs/design/baremetal/networking-infrastructure.md).
+
+In cluster network infrastructure automates a number
+of components that are handled on other platforms by cloud infrastructure services.
+
+### Load-balanced control plane access
+
+Access to the Kubernetes API (port 6443) from clients both external
+and internal to the cluster should be load-balanced across control
+plane machines.
+
+Access to Ignition configs (port 22623) from clients within the
+cluster should also be load-balanced across control plane machines.
+
+In both cases, the installation process expects these ports to be
+reachable on the bootstrap instance at first and then later on the
+newly-deployed control plane machines.
```

Comment:
that's technically not correct, the expectation is that these are load balanced to any one of them as long as it's healthy.

There is no first bootstrap and then control-plane, it's all healthy endpoints.

Reply:
@abhinavdahiya I removed this section from the original doc:

> 
> On other platforms (for example, see [the AWS UPI
> instructions](../../user/aws/install_upi.md)) an external
> load-balancer is required to be configured in advance in order to
> provide this access.

For bootstrap I am unable to find haproxy in use.  From my review the expectation is the VIP moves to the control plane node once the bootstrap node is destroyed.

cc: @russellb